lib/compiler.js

```diff
@@ -38,6 +38,8 @@ module.exports = function compile(options) {
 
   var appRootDir = options.appRootDir = options.appRootDir || process.cwd();
   var env = options.env || process.env.NODE_ENV || 'development';
+  var scriptExtensions = options.scriptExtensions = options.scriptExtensions ||
+                                                    _.keys(require.extensions);
```

Comment:
Considering that you are converting this  array back into an object down in `isPreferredExtension`, I think it's better to make the conversion right here.

```js
var scriptExtensions = options.scriptExtensions ?
  arrayToObject(scriptExtensions) : 
  require.extensions;
```